package.json

```diff
@@ -502,6 +504,22 @@
                     "default": null,
                     "description": "%makefile-tools.configuration.makefile.compileCommandsPath.description%",
                     "scope": "resource"
+                },
+                "makefile.optionalFeatures": {
```

Comment:
I think the old name `makefile.panel.visibility` was better.  "Optional features" is a bit too vague.  And if we copy the pattern from CMake Tools, the individual settings can have simpler names like "debug" and "run" because the context is already understood.

Reply:
I see your point. Let me put that back. Do you think I should have a type in configurations.ts that matches the `makefile.panel.visibility` exactly:
```
// as described in makefile.panel.visibility
type MakefilePanelVisibility = {
    enableLocalDebugging: boolean,
    enableLocalRunning: boolean
};
```
I am not sure what the conventions are. But having an easily relatable name for the type that represents the object that is encapsulated by makefile.panel.visibility makes sense to me.

Reply:
OK. Changed the property back to `makefile.panel.visibility` and adjusted types to be relatable to that name.

Reply:
I'll let @andreeis comment on that. She knows the code better than I do. I tend to chime in more on the stuff that our users would notice like settings, commands, and such.

Reply:
OK. Thanks.

Reply:
@andreeis , if there is any more work needed on this PR, please let me know. Thanks in advance for your input.